rskj-core/src/main/java/co/rsk/peg/whitelist/WhitelistSupportImpl.java

```diff
@@ -0,0 +1,209 @@
+     * @param addressBase58 the base58-encoded address to remove from the whitelist
+     * @return 1 SUCCESS, -1 ADDRESS_NOT_EXIST, -2 INVALID_ADDRESS_FORMAT, -10 GENERIC_ERROR,
+     * 0 UNKNOWN_ERROR.
+     */
+    @Override
+    public int removeLockWhitelistAddress(Transaction tx, String addressBase58) {
+        if (isAuthorizedLockWhitelistChange(tx)) {
+            return GENERIC_ERROR.getCode();
+        }
+
+        LockWhitelist whitelist = whitelistStorageProvider.getLockWhitelist();
+        try {
+            Address address = Address.fromBase58(btcContext.getParams(), addressBase58);
+
+            if (!whitelist.remove(address)) {
+                return ADDRESS_NOT_EXIST.getCode();
+            }
+            return SUCCESS.getCode();
+        } catch (AddressFormatException e) {
+            return INVALID_ADDRESS_FORMAT.getCode();
+        } catch (Exception e) {
+            logger.error("Unexpected error in removeLockWhitelistAddress: {}", e.getMessage());
+            panicProcessor.panic("lock-whitelist", e.getMessage());
+            return UNKNOWN_ERROR.getCode();
+        }
+    }
+
+    /**
+     * Sets a delay in the BTC best chain to disable lock whitelist
+     *
+     * @param tx                  current RSK transaction
+     * @param disableBlockDelayBI block since current BTC best chain height to disable lock
+     *                            whitelist
+     * @return 1 SUCCESS, -1 DELAY_ALREADY_SET, -2 DISABLE_BLOCK_DELAY_INVALID, -10 GENERIC_ERROR
+     */
+    @Override
+    public int setLockWhitelistDisableBlockDelay(Transaction tx, BigInteger disableBlockDelayBI, int btcBlockchainBestChainHeight)
+        throws IOException, BlockStoreException {
+
+        if (isAuthorizedLockWhitelistChange(tx)) {
+            return GENERIC_ERROR.getCode();
+        }
+
+        LockWhitelist lockWhitelist = whitelistStorageProvider.getLockWhitelist();
+        if (lockWhitelist.isDisableBlockSet()) {
+            return DELAY_ALREADY_SET.getCode();
+        }
+
+        int disableBlockDelay = disableBlockDelayBI.intValueExact();
+        if (disableBlockDelay + btcBlockchainBestChainHeight <= btcBlockchainBestChainHeight) {
+            return DISABLE_BLOCK_DELAY_INVALID.getCode();
+        }
+
+        lockWhitelist.setDisableBlockHeight(btcBlockchainBestChainHeight + disableBlockDelay);
+
+        return SUCCESS.getCode();
+    }
+
+    private Integer addLockWhitelistAddress(Transaction tx, LockWhitelistEntry whitelistEntry) {
+        if (isAuthorizedLockWhitelistChange(tx)) {
+            return GENERIC_ERROR.getCode();
+        }
+
+        LockWhitelist whitelist = whitelistStorageProvider.getLockWhitelist();
+        try {
+            if (whitelist.isWhitelisted(whitelistEntry.address())) {
+                return ADDRESS_ALREADY_WHITELISTED.getCode();
+            }
+            whitelist.put(whitelistEntry.address(), whitelistEntry);
+            return SUCCESS.getCode();
+        } catch (Exception e) {
```

Comment:
What's the possible exception here to be caught?

Reply:
Done! I removed it. This exception is not needed.